packages/core/src/errors/utxo.ts

```diff
@@ -0,0 +1,32 @@
+import { BaseError } from './base.js'
+
+export type UTXOsFetchErrorType = UTXOsFetchError & {
+  name: 'UTXOsFetchError'
+}
+
+export class UTXOsFetchError extends BaseError {
+  constructor({ address }: { address: string }) {
+    super(`There was a problem fetching UTXOs for address: ${address}.`, {
+      name: 'UTXOsFetchError',
+    })
+  }
+}
+
+export type NotEnoughUTXOsErrorType = NotEnoughUTXOsError & {
+  name: 'NotEnoughUTXOsError'
+}
+
+export class NotEnoughUTXOsError extends BaseError {
```

Comment:
How do you feel about renaming it to `InsufficientUTXOBalanceError`?

Reply:
Yeah, that sounds more appropriate,